### PR TITLE
Integration tests

### LINKED ISSUE #205: Need integration testing for all sample applications
Need to add `make bootc` testing for:
(pending #254)
- [x] codegen
- [x] summarizer
- [x] rag
- [x] audio-to-text

### PR DESCRIPTION
**Pull request by sallyom**
Depends on #247 
This adds integration tests & pushes/mirrors new images to quay.io/ai-lab
Resolves https://github.com/containers/ai-lab-recipes/issues/205

### CODE REVIEW

**Comment by sallyom** [comment]
@Gregory-Pereira @lmilbaum this is failing with "out of VPCs" error, specific to AWS account - this adds `make bootc` test for various apps in the testing-framework workflow. 

```
Error: creating EC2 VPC: operation error EC2: CreateVpc, https response error StatusCode: 400, api error VpcLimitExceeded: The maximum number of VPCs has been reached.
```

**Comment by MichaelClifford** [comment]
@sallyom is this PR still needed?

File: .github/workflows/testing-framework.yaml [hunk 1]
```diff
@@ -139,8 +145,14 @@ jobs:
       matrix:
         include:
           - image: llamacpp_python
+          #- image: llamacpp_python_vulkan
+          - image: llamacpp_python_cuda
```
Comment (Gregory-Pereira) [inline .github/workflows/testing-framework.yaml:149]: These are labeled wrong, `llamacpp_python` is correct though:
```suggestion
          #- image: llamacpp-python-vulkan
          - image: llamacpp-python-cuda
```

**Review by Gregory-Pereira — COMMENTED** [review]
/lgtm, but 1 change

**Comment by Gregory-Pereira** [comment]
@lmilbaum these are failing due to max VPCs in AWS [it looks like](https://github.com/containers/ai-lab-recipes/actions/runs/8677265648/job/23792771867?pr=254#step:6:550). We would probably have to increase that if we want integration tests for all of our applications: `VpcLimitExceeded: The maximum number of VPCs has been reached.`. Also will we want to run these jobs on the `containers-mirror/ai-lab-recipes` or `containers/ai-lab-recipes`.

**Comment by lmilbaum** [comment]
@sallyom Should this contribution land in 1.1.0 release?

**Comment by Gregory-Pereira** [comment]
I don't know, I was never informed that this is required. That being said Sally has been working on this for a while, I say we defer to her judgement here. If she wants it we can work with here to see if we can get in.

**Comment by rhatdan** [comment]
I would say no, this should not block release.

**Comment by lmilbaum** [comment]
Is this PR still needed? It is stale for quite some time now.

**Comment by rhatdan** [comment]
I will close it, @sallyom can reopen it of create a new PR.

**Pull request closed without merging**